common/src/main/java/ch/admin/bag/covidcertificate/common/qr/QrScanFragment.kt

```diff
@@ -160,60 +235,32 @@ abstract class QrScanFragment : Fragment() {
 		}
 	}
 
-	private fun startCameraAndQrAnalyzer() {
-		barcodeScanner.apply {
-			val formats: Collection<BarcodeFormat> = listOf(BarcodeFormat.AZTEC, BarcodeFormat.QR_CODE)
-			barcodeView.decoderFactory = DefaultDecoderFactory(formats)
-			decodeContinuous(callback)
-			setStatusText("")
-			viewFinder.visibility = View.GONE
-		}
-		val cameraSettings = barcodeScanner.cameraSettings
-		cameraSettings.focusMode
-		setupFlashButton()
-	}
-
 	private fun setupFlashButton() {
-		barcodeScanner.setTorchListener(object : DecoratedBarcodeView.TorchListener {
-			override fun onTorchOn() {
-				this@QrScanFragment.isTorchOn = true
-				setupFlashButtonStyle()
-			}
+		val camera = camera ?: return
 
-			override fun onTorchOff() {
-				this@QrScanFragment.isTorchOn = false
-				setupFlashButtonStyle()
-			}
-
-		})
+		if (camera.cameraInfo.hasFlashUnit()) {
+			flashButton.isVisible = true
+			setFlashAndButtonStyle()
+		} else {
+			flashButton.isVisible = false
+		}
 
 		flashButton.setOnClickListener {
-			val isOn = flashButton.isSelected
-			if (isOn) {
-				barcodeScanner.setTorchOff()
-			} else {
-				barcodeScanner.setTorchOn()
-			}
+			isTorchOn = !flashButton.isSelected
+			setFlashAndButtonStyle()
 		}
 	}
 
-	private fun setupFlashButtonStyle() {
-		if (isTorchOn) {
-			flashButton.apply {
-				isSelected = true
-				setImageDrawable(ContextCompat.getDrawable(context, torchOnDrawable))
-			}
-		} else {
-			flashButton.apply {
-				isSelected = false
-				setImageDrawable(ContextCompat.getDrawable(context, torchOffDrawable))
-			}
-		}
+	private fun setFlashAndButtonStyle() {
+		camera?.cameraControl?.enableTorch(isTorchOn)
+		val drawableId = if (isTorchOn) torchOnDrawable else torchOffDrawable
+		flashButton.isSelected = isTorchOn
+		flashButton.setImageResource(drawableId)
 	}
 
 	private fun handleInvalidQRCodeExceptions(qrCodeData: String?, error: Error?) {
+		//TODO maybe
```

Comment:
TO be DOne when?

Reply:
Before the next release, but we need this pull request for the testing session